From 060cbe52e3b68f49e3f44b0324e17c3e8533d1ca Mon Sep 17 00:00:00 2001
From: WANG Xuerui <git@xen0n.name>
Date: Wed, 1 Sep 2021 14:16:21 +0800
Subject: [PATCH 5/8] syscalls: add build system support for syncing
 loongarch64 syscalls

This is incomplete because the syscalls-loongarch64.txt has not been
imported yet, and will not build. This file will be imported in the next
commit.
---
 src/basic/meson.build         | 1 +
 src/basic/missing_syscalls.py | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/src/basic/meson.build b/src/basic/meson.build
index 452b965db3..e4b6720fa3 100644
--- a/src/basic/meson.build
+++ b/src/basic/meson.build
@@ -330,6 +330,7 @@ arch_list = [
         'arm64',
         'i386',
         'ia64',
+        'loongarch64',
         'm68k',
         'mips64',
         'mips64n32',
diff --git a/src/basic/missing_syscalls.py b/src/basic/missing_syscalls.py
index dd458994f2..2694e83b98 100644
--- a/src/basic/missing_syscalls.py
+++ b/src/basic/missing_syscalls.py
@@ -60,6 +60,8 @@ DEF_TEMPLATE_B = '''\
 #    define systemd_NR_{syscall} {nr_i386}
 #  elif defined(__ia64__)
 #    define systemd_NR_{syscall} {nr_ia64}
+#  elif defined(__loongarch64)
+#    define systemd_NR_{syscall} {nr_loongarch64}
 #  elif defined(__m68k__)
 #    define systemd_NR_{syscall} {nr_m68k}
 #  elif defined(_MIPS_SIM)
-- 
2.35.1

